File: Python3/Array&String/1431. Kids With the Greatest Number of Candies.py
""" 1431. Kids With the Greatest Number of Candies
Difficulty: Easy
Submission Link: https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/submissions/1017418134
 """

class Solution:
    def kidsWithCandies(self, candies: list[int], extraCandies: int) -> list[bool]:
        max_candies = max(candies)
        result = []
        
        for candies_count in candies:
            result.append(candies_count + extraCandies >= max_candies)
        
        return result
    

File: C++/Array&String/1431. Kids With the Greatest Number of Candies.cpp
// 1431. Kids With the Greatest Number of Candies
// Difficulty: Easy
// Submission Link: https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/submissions/1017415796

class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int maxCandies = *max_element(candies.begin(), candies.end());
        vector<bool> result;
        
        for (int i = 0; i < candies.size(); i++) {
            if (candies[i] + extraCandies >= maxCandies) {
                result.push_back(true);
            } else {
                result.push_back(false);
            }
        }
        
        return result;
    }
};